packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -333,10 +342,21 @@ Future<String> _initializeGradle(FlutterProject project) async {
     );
   } on ProcessException catch (e) {
     final String error = e.toString();
-    if (error.contains('java.io.FileNotFoundException: https://downloads.gradle.org') ||
-        error.contains('java.io.IOException: Unable to tunnel through proxy')) {
+    if (_kKnownErrorPrefixes.any((String candidate) => error.contains(candidate))) {
       throwToolExit('$gradle threw an error while trying to update itself.\n$e');
     }
+    // gradlew is missing execute.
+    if (error.contains('Permission denied')) {
+      throwToolExit(
+        '$gradle does not have permission to execute by your user.\n'
+        'You should change the ownership of these directories to your user, or '
```

Comment:
Can we say anything more specific than 'these directories' ?

Reply:
Specified "project directory", but we do show the path to the gradle above. I'll update it to include the original error.